src/katgpucbf/utils.py

```diff
@@ -68,14 +68,16 @@ class DeviceStatus(Enum):
     FAIL = 3
 
 
-class DeviceStatusSensor(aiokatcp.AggregateSensor[DeviceStatus]):
+class DeviceStatusSensor(aiokatcp.SimpleAggregateSensor[DeviceStatus]):
     """Summary sensor for quickly ascertaining device status.
 
     This takes its value from the worst status of its target set of sensors, so
     it's quick to identify if there's something wrong, or if everything is good.
     """
 
     def __init__(self, target: aiokatcp.SensorSet) -> None:
+        # We count the number of sensors with each possible status
+        self._counts: Counter[aiokatcp.Sensor.Status] = Counter()
```

Comment:
I did vaguely think of keeping lists / counters / some kind of track of sensors in various states. It seemed more complicated to do it this way though, not less.